cnxpublishing/views/admin.py

```diff
@@ -97,6 +125,241 @@ def admin_post_publications(request):
+                latest_recepie = row[4]
+                current_recepie = row[5]
+                latest_version = row[6]
+                current_version = row[7]
+                state = str(result.state)
+                if current_version != latest_version:
+                    state += ' stale_content'
+                if current_recepie != latest_recepie:
+                    state += ' stale_recipie'
+                state_icon = result.state
+                if state[:7] == "SUCCESS" and len(state) > 7:
+                    state_icon = 'PENDING'
+                states.append({
+                    'title': row[0].decode('utf-8'),
+                    'authors': format_autors(row[1]),
+                    'uuid': row[2],
+                    'print_style': row[3],
+                    'recipe': row[4],
+                    'created': row[-2],
+                    'state': state,
+                    'state_message': message,
+                    'state_icon': STATE_ICONS[state_icon]['class'],
+                    'state_icon_style': STATE_ICONS[state_icon]['style']
+                })
+    status_filters = [request.GET.get('pending_filter', ''),
+                      request.GET.get('started_filter', ''),
+                      request.GET.get('retry_filter', ''),
+                      request.GET.get('failure_filter', ''),
+                      request.GET.get('success_filter', '')]
+    status_filters = [x for x in status_filters if x != '']
+    if status_filters == []:
+        status_filters = ["PENDING", "STARTED", "RETRY", "FAILURE", "SUCCESS"]
+    for f in (status_filters):
+        args[f] = "checked"
+    final_states = []
+    for state in states:
+        if state['state'].split(' ')[0] in status_filters:
+            final_states.append(state)
+
+    sort = request.GET.get('sort', 'bpsa.created DESC')
+    sort_match = SORTS_DICT[sort.split(' ')[0]]
+    args['sort_' + sort_match] = ARROW_MATCH[sort.split(' ')[1]]
+    args['sort'] = sort
+    if sort == "STATE ASC":
+        final_states = sorted(final_states, key=lambda x: x['state'])
+    if sort == "STATE DESC":
+        final_states = sorted(final_states,
+                              key=lambda x: x['state'], reverse=True)
+
+    num_entries = request.GET.get('number', 100)
+    page = request.GET.get('page', 1)
+    try:
+        page = int(page)
+        num_entries = int(num_entries)
+        start_entry = (page - 1) * num_entries
+    except ValueError:
+        raise httpexceptions.HTTPBadRequest(
+            'invalid page({}) or entries per page({})'.
+            format(page, num_entries))
+    final_states = final_states[start_entry: start_entry + num_entries]
+    args.update({'start_entry': start_entry,
+                 'num_entries': num_entries,
+                 'page': page,
+                 'total_entries': len(final_states)})
+
+    args.update({'states': final_states})
+    return args
```

Comment:
This is a pretty long function. And by the end I'm not sure what is being returned other than it's a dictionary called 'args'. Any change we could create this dictionary at the very just before or in  the return statement? This would allow the casual observer to know exactly what the response structure from this function will look like.